packages/svelte/src/internal/client/types.d.ts

```diff
@@ -30,28 +30,39 @@ export type Store<V> = {
 	set(value: V): void;
 };
 
+// For all the core internal objects, we use signal character property strings.
+// This not only reduces code-size and parsing, but it also improves the performance
+// when the JS VM JITs the code.
+
 export type ComponentContext = {
-	props: MaybeSignal<Record<string, unknown>>;
-	accessors: Record<string, any> | null;
-	effects: null | Array<EffectSignal>;
-	mounted: boolean;
-	parent: null | ComponentContext;
-	context: null | Map<unknown, unknown>;
-	immutable: boolean;
-	runes: boolean;
-	update_callbacks: null | {
-		before: Array<() => void>;
-		after: Array<() => void>;
-		execute: () => void;
+	// props
```

Comment:
Can you make all these jsdoc comments? That way you can see it when hovering over the property on the usage site. You can then also remove the corresponding comments where you're instantiating these shapes in their factory functions.

```suggestion
	/** props */
```

Reply:
Good point.